server.js

```diff
@@ -27,6 +30,51 @@ app.get('/', function (req, res) {
      res.sendFile(__dirname + '/src/index.html');
 });
 
+//Marketing endpoint
+app.get('/marketing', function(req,res){
+    const con = mysql.createConnection(databaseConnectionOptions);
+    var today= moment.utc();
+    //Starting date of this week
+    var thisWeek = today.startOf("week");
+    //date from 2 weeks ago
+    var twoWeeks = thisWeek.clone().subtract(2,"week");
+    //this month start date
+    var thisMonth = today.startOf("month");
+    //last month start date
+    var lastMonth = thisMonth.clone().subtract(1, "month");
+    // //Get all registrations
+    var query = "SELECT COUNT(*) as total FROM str_register; ";
+    query+= `SELECT COUNT(*) as total FROM str_register WHERE created_at >= '${thisWeek.format("YYYY-MM-DD hh:mm:ss")}';`;
+    query+= `SELECT COUNT(*) as total FROM str_register WHERE created_at >= '${twoWeeks.format("YYYY-MM-DD hh:mm:ss")}' AND created_at < '${thisWeek.format("YYYY-MM-DD hh:mm:ss")}';`;
+    query+= `SELECT COUNT(*) as total FROM str_register WHERE created_at >= '${thisMonth.format("YYYY-MM-DD hh:mm:ss")}';`;
+    query+= `SELECT COUNT(*) as total FROM str_register WHERE created_at >= '${lastMonth.format("YYYY-MM-DD hh:mm:ss")}' AND created_at < '${thisMonth.format("YYYY-MM-DD hh:mm:ss")}';`;
+
+    con.query(query,(err,results)=>{
+        try{
+            if(err){res.send(err);return;}
+            resultArray = JSON.parse(JSON.stringify(results));
+            totals = {
+                total : resultArray[0][0].total,
+                totalWeek : resultArray[1][0].total,
+                totalTwoWeeks : resultArray[2][0].total,
+                totalMonth : resultArray[3][0].total,
+                totalLastMonth : resultArray[4][0].total
+                }
+            res.status(200);
+            res.send(totals);
+            
+        }catch(error){
+            console.warn(new Error(error));
```

Comment:
Send back an error status in the situation that something goes wrong so that the frontend can fail gracefully